Add trajectory preview subtree for convenience

pac48 opened the pull request:
This PR address https://github.com/PickNikRobotics/moveit_studio/issues/3545
I added a new objective to the `picknik_ur_base_config` package to preview a trajectory if the user is available.

sea-bass commented on src/picknik_ur_base_config/objectives/wait_for_trajectory_approval_if_user_available.xml:
```diff
@@ -0,0 +1,11 @@
+      <Inverter>
+        <Action ID="IsUserAvailable"/>
+      </Inverter>
+      <Action ID="WaitForUserTrajectoryApproval" solution="{mtc_solution}"/>
```
```suggestion
      <Action ID="WaitForUserTrajectoryApproval" solution="{solution}"/>

sea-bass commented on src/picknik_ur_base_config/objectives/wait_for_trajectory_approval_if_user_available.xml:
```diff
@@ -0,0 +1,11 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<root BTCPP_format="4" main_tree_to_execute="Wait for Trajectory Approval if User Available">
+  <BehaviorTree ID="Wait for Trajectory Approval if User Available" _description="Preview a shared pointer to an MTC Solution object via an input data port if a user is available.">
+    <Fallback name="wait_for_approval_if_user_available">
+      <Inverter>
+        <Action ID="IsUserAvailable"/>
+      </Inverter>
+      <Action ID="WaitForUserTrajectoryApproval" solution="{mtc_solution}"/>
+    </Fallback>
```
We actually use the complete syntax here.

```suggestion
    <Control ID="Fallback" name="wait_for_approval_if_user_available">
      <Decorator ID="Inverter">
        <Action ID="IsUserAvailable"/>
      </Decorator>
      <Action ID="WaitForUserTrajectoryApproval" solution="{mtc_solution}"/>
    </Control>
```

sea-bass reviewed (changes requested):
The subtree looks great -- but I think we should also use it as a subtree in all the Objectives that currently use it!

For example, in [this objective](https://github.com/PickNikRobotics/moveit_studio_ws/blob/main/src/picknik_ur_site_config/objectives/pick_object.xml#L15-L20).

old:
```
            <Fallback name="wait_for_approval_if_user_available">
                <Inverter>
                    <Action ID="IsUserAvailable"/>
                </Inverter>
                <Action ID="WaitForUserTrajectoryApproval" solution="{pick_object_solution}"/>
            </Fallback>
            <Action ID="ExecuteMTCTask" solution="{pick_object_solution}"/>
```

new:
```
            <SubTree ID="Wait for Trajectory Approval if User Available" solution="{mtc_solution}">
```

It should end up looking something like this:

![image](https://github.com/PickNikRobotics/moveit_studio_ws/assets/4603398/429a5c1f-624e-4c60-accc-ac2d23cad193)

pac48 commented:
> The subtree looks great -- but I think we should also use it as a subtree in all the Objectives that currently use it!
> 
> For example, in [this objective](https://github.com/PickNikRobotics/moveit_studio_ws/blob/main/src/picknik_ur_site_config/objectives/pick_object.xml#L15-L20).
> 
> old:
> 
> ```
>             <Fallback name="wait_for_approval_if_user_available">
>                 <Inverter>
>                     <Action ID="IsUserAvailable"/>
>                 </Inverter>
>                 <Action ID="WaitForUserTrajectoryApproval" solution="{pick_object_solution}"/>
>             </Fallback>
>             <Action ID="ExecuteMTCTask" solution="{pick_object_solution}"/>
> ```
> 
> new:
> 
> ```
>             <SubTree ID="Wait for Trajectory Approval if User Available" solution="{mtc_solution}">
> ```
> 
> It should end up looking something like this:
> 
> ![image](https://private-user-images.githubusercontent.com/4603398/291461483-429a5c1f-624e-4c60-accc-ac2d23cad193.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI5OTU1OTcsIm5iZiI6MTcwMjk5NTI5NywicGF0aCI6Ii80NjAzMzk4LzI5MTQ2MTQ4My00MjlhNWMxZi02MjRlLTRjNjAtYWNjYy1hYzJkMjNjYWQxOTMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQUlXTkpZQVg0Q1NWRUg1M0ElMkYyMDIzMTIxOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyMzEyMTlUMTQxNDU3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZWUzM2I4YzRlMDE4Y2Y0NmI2N2Y5YzlhNzE0NDI5NzczMGY0MzVmMzFiNWFlOTU2ZDkxZWYzNDUxNDkxOTk2ZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.tGhc-leFoLE9sQYOlay3pI2ECpLa3FFuKdmVH95o6FM)

Should the new block be like this instead? I don't see where {mtc_solution} would be referenced from.
```            
<SubTree ID="Wait for Trajectory Approval if User Available" solution="{pick_object_task}">
```

sea-bass commented:
> Should the new block be like this instead? I don't see where {mtc_solution} would be referenced from.
> 
> ```
> <SubTree ID="Wait for Trajectory Approval if User Available" solution="{pick_object_task}">
> ```

Yes, good call, but it should be `"{pick_object_solution}"` based on what I see.

pac48 commented:
> > Should the new block be like this instead? I don't see where {mtc_solution} would be referenced from.
> > ```
> > <SubTree ID="Wait for Trajectory Approval if User Available" solution="{pick_object_task}">
> > ```
> 
> Yes, good call, but it should be `"{pick_object_solution}"` based on what I see.

Great, I added `"pick_object_solution"` and fixed the rest.
